volatility3/cli/volargparse.py

```diff
@@ -71,6 +75,24 @@ def __call__(
             getattr(namespace, argparse._UNRECOGNIZED_ARGS_ATTR).extend(arg_strings)
 
 
+def find_parser(parser_name: str, parsers: Dict[str, PluginInterface]) -> List[str]:
+    """Attempt to find parsers for the given parser name."""
+
+    # If the last part of the provided module path is the same as the part before it, this is likely a legacy
+    # plugin name invocation. For example 'windows.devicetree.Devicetree' is translated to 'windows.devicetree'.
+
+    parser_name = parser_name.lower()
+    parts = parser_name.split(".")
+
+    if len(parts) >= 2 and parts[-1].replace("_", "") == parts[-2].replace("_", ""):
+        parser_name = parser_name.rpartition(".")[0]
+
+    partial_matches = [name for name in parsers if parser_name in name.lower()]
+    exact_matches = [name for name in parsers if parser_name == name.lower()]
+
+    return exact_matches if exact_matches else partial_matches
```

Comment:
This leads to issues give the "compression" is selective and technically a module could contain a plugin with its name *and* with another name.  So I think these changes need to go, sadly.